### PR TITLE
GH-370: Enable ScriptProcessor tests back

### PR DESCRIPTION
**Pull request by artembilan**
Fixes https://github.com/spring-cloud/stream-applications/issues/370

* Update dependencies for `script-processor`
* Add `maven-surefire-plugin` to `script-processor` POM and specify `--add-opens` for Ruby requirements
* Mark all JavaScript tests with `@EnabledIfSystemProperty` against `org.graalvm.language.js.home` which is available only if GraalVM with `js` component enabled
* Fix `NewlineAtEndOfFile` rule for Checkstyle since on Windows the line end is `crlf`
* Add `components: 'js'` to GraalVM installation steps for all the GH action scripts

### CODE REVIEW

**Review by onobc — APPROVED** [review]
Great work @artembilan - LGTM.

**Comment by artembilan** [comment]
@corneil ,
I see my PR has failed:
```
/runner/_work/_temp/584d1886-6a09-43a0-a6ac-f04d36cf62b3.sh: line 4: gh: command not found
Error: Process completed with exit code 127.
```
It points to this line of code in the `ci-pr.yml`:
```
gh pr diff --name-only
```
Have no idea what is that and I doubt it related somehow to my changes...
Feels like something else has to installed for that `shell: bash` step.

Thanks for understanding

**Comment by onobc** [comment]
> @corneil , I see my PR has failed:
> 
> ```
> /runner/_work/_temp/584d1886-6a09-43a0-a6ac-f04d36cf62b3.sh: line 4: gh: command not found
> Error: Process completed with exit code 127.
> ```
> 
> It points to this line of code in the `ci-pr.yml`:
> 
> ```
> gh pr diff --name-only
> ```
> 
> Have no idea what is that and I doubt it related somehow to my changes... Feels like something else has to be installed for that `shell: bash` step.
> 
> Thanks for understanding

Ditto here @corneil https://github.com/spring-cloud/stream-applications/actions/runs/3888288703

**Comment by artembilan** [comment]
Pushed the fix with extra dependencies which make working JavaScript on a regular JVM.
No strong GraalVM requirements any more!

File: .github/workflows/ci-pr.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
         with:
           version: 'latest'
           java-version: '17'
-          #          components: 'native-image' # add when starting native builds.
+          components: 'js'
```
[the remark this answers is not in the document]
Comment (artembilan) [reply .github/workflows/ci-pr.yml:37]: It won't hurt and won't need to bring it back when we start building native images for our applications.

**Review by onobc — APPROVED** [review]
LGTM @artembilan

**Pull request merged**